mos-platform/pet/link.ld

```diff
@@ -0,0 +1,19 @@
+/* Commodore PET PRG file linker script.                                                                 
+ *                                                                                                       
+ * Produces a PRG file with a SYS command to start the program.                                          
+ */
+
+MEMORY {
+    ram (rw) : ORIGIN = 0x0401, LENGTH = 0x7BFF
```

Comment:
You can use expressions involving symbols to set the length of the RAM; that allows users to customize the layout just by defining a symbol, either in a `.s` file or inline assembly.

The NES targets currently do this; the convention they've used is to take the size of various regions in kilobytes, then do the necessary math based on this. Here, that would be the calculations to set the `LENGTH` of `ram` and the value of `__stack`. You'd also probably want an [`ASSERT` statement](https://sourceware.org/binutils/docs/ld/Miscellaneous-Commands.html) to ensure that the value of the symbol is in range. Finally, you can [`PROVIDE`](https://sourceware.org/binutils/docs/ld/PROVIDE.html) the symbol a default value in case the user doesn't supply one.

Reply:
This should be implemented now, I ran some tests on VICE on lower memory models which fail spectacularly with this version but now work with the inline assembly defining the right memory size.

Reply:
Looks good!